active-rfcs/0000-finalize-head-body.md

```diff
@@ -0,0 +1,80 @@
+- Start Date: 11/30/2021
+- Reference Issues: https://github.com/withastro/rfcs/discussions/15
+- Implementation PR:
+
+# Summary
+
+Finalize behavior around `<head>`, `<body>`, `<html>`, and `<!DOCTYPE html>` elements in Astro component templates.
+
+# Motivation
+
+in Astro v0.21, we have had user reports around `<head>` not acting as expected, breaking some projects from v0.20:
+- https://github.com/withastro/astro/issues/2046
+- https://github.com/withastro/astro/issues/2022
+
+Some of these issues stem from undocumented or undefined behaviors in v0.20: was it allowed for `<head>` be conditionally added in a nested expression? 
+
+Other issues stem from our attempt to provide implicit `<head>` and `<body>` support in pages and layouts. There was a quick workaround added for v0.21 where `src/pages` and `src/layouts` folder locations must be defined in config so that our parser knows how to parse them as full HTML documents and not smaller HTML fragments. This unblocked users but at the expense of breaking two larger design goals of the project:
+  - Added `src/layouts` as a new special folder location, when only `src/pages` was intented to be "special".
+  - Caused Astrp to create different output for a component based on the file path of that component.
+
+## Goals
+
+1. Agree on and document the rules and behaviors of the `<head>`, `<body>`, `<html>`, and `<!DOCTYPE html>` tags.
+2. Remove special compiler behavior based on where a file lives in your src directory. 
+3. Remove `src/layouts` as a "special" folder and any config around this.
+
+## Non-Goals
+
+1. **Out of scope:** Support `<head>` Injection inside of individual components. This is a commonly requested user feature, but can be tackled as an additional feature, leaving this RFC focused on clarifying our existing API. This RFC does not impact that feature request.
+
+
+# Detailed design
+
+## Template Changes
+
+- `<!DOCTYPE html>`
+  - compiler: completely ignored by the compiler, stripped out from render output.
+  - runtime: a `<!DOCTYPE html>` tag is always included in final page HTML output.
+- `<html>`, `<body>`, `<head>`
+  - compiler: will be left as-is in the component template.
+  - runtime: will be left as-is in final page HTML output.
+  - runtime: may warn if duplicate `html`, `body`, and `head` tags are included in final page HTML output.
```

Comment:
this may be difficult, remembering that the runtime only see strings of HTML, so not easy to detect multiple tags.

Reply:
I can it make more clear that this isn't required, and is just a suggestion of something that is not impossible/blocked by this RFC.